net-core/Ical.Net/Ical.Net/DataTypes/WeekDay.cs

```diff
@@ -64,6 +66,20 @@ public override void CopyFrom(ICopyable obj)
             }
         }
 
+        public int CompareTo(WeekDay other)
+        {
+            if (other == null)
+            {
+                throw new ArgumentException();
+            }
+            var compare = DayOfWeek.CompareTo(other.DayOfWeek);
+            if (compare == 0)
+            {
+                compare = Offset.CompareTo(other.Offset);
+            }
+            return compare;
+        }
+
         public int CompareTo(object obj)
```

Comment:
Let's get rid of this `IComparable` implementation in favor of just `IComparable<T>`, so:
* Delete this method
* Delete `IComparable` from `IWeekDay`

(This may only apply to v2.)